Useful software
=======

Catagory（update: 2017-09-11） | Name
--- | ---
**Gene expression** |
Gene expression across species and biological conditions | [Expression Atlas](https://www.ebi.ac.uk/gxa/home)
A comprehensive framework for accurate and efficient RNA-Seq analysis | [RNACocktail](https://bioinform.github.io/rnacocktail/)
a comprehensive and up-to-date set of non-coding RNA sequences | [RNAcentral](http://rnacentral.org)
The genomic tRNA database contains tRNA gene predictions made by tRNAscan-SE on complete or nearly complete genomes. | [GtRNAdb](http://gtrnadb.ucsc.edu/)
|
**MicroRNA** | 
miRNA target prediction and functional annotations | [miRDB](http://mirdb.org/miRDB/)
Predicted microRNA targets & target downregulation scores | [miRanda](http://www.microrna.org/microrna/home.do)
MicroRNA target prediction tools | [Catalog](http://www.exiqon.com/microrna-target-prediction)
microRNA target detection | [RNA22](https://cm.jefferson.edu/rna22v2/)
a supervised machine learning approach for microRNA target prediction | [TargetSpy](http://webclu.bio.wzw.tum.de/targetspy/index.php?down=true)
target accessibility is a critical factor in microRNA function | [PITA](https://genie.weizmann.ac.il/pubs/mir07/index.html)
a comprehensive pipeline analysis suite designed for mRNA, miRNA and circRNA identification and differential expression analysis, applicable to any sequenced organism. | [miARma-Seq](http://miarmaseq.cbbio.es/)
elucidate the small RNA contents of deep sequencing reads | [smallrnaseq](https://github.com/dmnfarrell/smallrnaseq/wiki)
collection of several tools for RNA analysis | [sRNAtoolBox](http://bioinfo2.ugr.es:8080/srnatoolbox/)
|
**Genetics** | 
Online Mendelian Inheritance in Man | [OMIM](https://omim.org/)
genomic variation and its relationship to human health | [ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/)
a wiki investigating human genetics | [SNPedia](http://www.snpedia.com/)
The Human Gene Mutation Database | [HGMD](http://www.hgmd.cf.ac.uk/)
SNP regulatory annotation DB | [RegulomeDB](http://regulome.stanford.edu/)
effects of genetic variation on human health | [GHR](https://ghr.nlm.nih.gov/)
**Diseases** |
The human disease database | [MalaCards](http://www.malacards.org/)
|
**Genomics** | 
A Modular, Open-Source whole genome assembler | [AMOS](http://amos.sourceforge.net)
Assembler designed for high-noise single-molecule sequencing | [Canu](https://canu.readthedocs.io/en/latest/)
NGS reads de novo assembler | [SOAPdenovo2](https://github.com/aquaskyline/SOAPdenovo2)
A Genomic Mapping and Alignment Program for mRNA and EST | [GMAP](http://research-pub.gene.com/gmap/)
Complete and accurate shotgun assembly of bacterial sized genomes | [HGAP](https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/HGAP)
Rapid bacterial SNP calling and core genome alignments | [Snippy](https://github.com/tseemann/snippy)
|
**Cancer genomics** |
Reliable and accurate identification of somatic point mutations in high-throughput sequencing data of cancer genomes | [MuTect](https://gatkforums.broadinstitute.org/gatk/categories/mutect)
Variant calling and somatic mutation/CNV detection for next-generation sequencing data | [varscan](https://dkoboldt.github.io/varscan/)
|
**Metagenomic** |
Fast, reliable protein-coding **gene prediction** for prokaryotic genomes | [prodigal](https://github.com/hyattpd/Prodigal)
Taxonomic sequence classifier | [Kraken](https://ccb.jhu.edu/software/kraken/)
rapid prokaryotic **genome annotation** | [Prokka](https://github.com/tseemann/prokka)
A metagenomic and isolate assembly and analysis pipeline built with AMOS | [metAMOS](http://marbl.github.io/metAMOS)
An ultra-fast single-node solution for large and complex metagenomics assembly | [MEGAHIT](https://github.com/voutcn/megahit)
|
**Pan genomics** | 
Roary the pan genome pipeline | [Roary](https://github.com/sanger-pathogens/Roary)
Calculate the assocations between all genes in the accessory genome and the traits | [Scoary](https://github.com/AdmiralenOla/Scoary)
Interactive visualization of bacterial pan-genomes | [FriPan](https://github.com/drpowell/FriPan)
|
**NGS quality control** |
A quality control tool for high throughput sequence data | [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
Automatic Filtering, Trimming, Error Removing and Quality Control for fastq data | [AfterQC](https://github.com/OpenGene/AfterQC)
Aggregate results from bioinformatics analyses across many samples into a single report | [MultiQC](http://multiqc.info/)
**Microbe** | 
Virulence factors of bacterial pathogens | [VFDB](http://www.mgc.ac.cn/VFs/main.htm)
Comprehensive Antibiotic Resistance Database | [CARD](https://card.mcmaster.ca/home)
pathogen-host interactions | [PHI-base](http://www.phi-base.org/index.jsp)
|
**Web tools** |
Web IDE | [Codiad](http://codiad.com/)
Gene Annotation Query as a Service | [MyGene.info](http://mygene.info/)
Versatile matrix visualization and analysis software | [MORPHEUS](https://software.broadinstitute.org/morpheus/)
Paletton Live Colorizer | [Paletton](http://paletton.com)
Venn plot | [Venny](http://bioinfogp.cnb.csic.es/tools/venny/index.html)
|
**Local tools** |
A modern, open source text editor that understands web design | [Brackets](http://brackets.io/)
open source bibliography reference manager | [JabRef](https://www.jabref.org/)

